orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/StageBuilderProvider.groovy

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.orca.batch
+
+public interface StageBuilderProvider {
+  Collection<StageBuilder> all()
```

Comment:
The `StageBuilderProvider` is to be used in all situations where we were doing a `getBeansOfType(StageBuilder)`.

It's capable of returning legacy `StageBuilder` and adapted `StageDefinitionBuilder`.